### PR TITLE
[1442] sts endpoint fix for china regions

### PR DESCRIPTION
**Pull request by narasimman-elu**
The pull request for issue #1442 
This fixes the sts endpoind issue for china provided the AWS_REGION environment shall be set by pod configs.
The fix is based on v7.0.7

### CODE REVIEW

File: pkg/credentials/iam_aws.go [hunk 1]
```diff
@@ -82,7 +83,11 @@ func (m *IAM) Retrieve() (Value, error) {
 	case len(os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")) > 0:
 		if len(endpoint) == 0 {
 			if len(os.Getenv("AWS_REGION")) > 0 {
-				endpoint = "https://sts." + os.Getenv("AWS_REGION") + ".amazonaws.com"
+				if strings.HasPrefix(os.Getenv("AWS_REGION"), "cn") {
```
Comment (vadmeste) [inline pkg/credentials/iam_aws.go:86]: `cn-` instead of `cn`

Comment (vadmeste) [reply pkg/credentials/iam_aws.go:86]: By the way, normally you can pass endpoint using NewIAM(endpoint string) API, that would be enough

Comment (narasimman-elu) [reply pkg/credentials/iam_aws.go:86]: Thanos is one of the client neither set endpoint while creeating IAM and nor using NewIAM function.
May be Thanos should consider to fix this issue also from Thanos side.

Comment (narasimman-elu) [reply pkg/credentials/iam_aws.go]: > `cn-` instead of `cn`

Have considered this review point and fixed.

**Review by poornas — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**